src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs

```diff
@@ -50,17 +47,25 @@ public WebSocketsTransport(HttpConnectionOptions httpConnectionOptions, ILoggerF
                 // Set an alternative user agent header on Full framework
                 _webSocket.Options.SetRequestHeader("X-SignalR-User-Agent", Constants.UserAgentHeader.ToString());
 #endif
+            }
 
-                if (httpConnectionOptions != null)
+            if (httpConnectionOptions != null)
+            {
+                if (httpConnectionOptions.Headers != null)
                 {
-                    if (httpConnectionOptions.Headers != null)
+                    if (isBrowser)
                     {
-                        foreach (var header in httpConnectionOptions.Headers)
-                        {
-                            _webSocket.Options.SetRequestHeader(header.Key, header.Value);
-                        }
+                        throw new PlatformNotSupportedException("Configuring request headers is not supported when using WebSocketsTransport in the browser platform.");
```

Comment:
This is a breaking change

Reply:
I would like to log a warning in this case though. Or rather when `httpConnectionOptions.Headers.Count > 0`. Like ClientCertificates, Headers cannot be set to null. It never starts out as null either.

I would prefer throwing if it weren't for the fact that someone might intentionally be configuring headers just for the negotiate request.